joint_calling/upload_processor.py

```diff
@@ -0,0 +1,110 @@
+""" This function prepares gVCF's uploaded to GCS, based off sample status
+    logged in a database, for further QC and downstream analysis.
+    The upload processor will determine when samples should be added
+    to existing MatrixTables where appropriate and which MatrixTables
+    they should be combined with in this case. Following a successful
+    run all uploaded files will be moved to archival storage.
+
+    Assumptions
+    ===========
+    - Specified files must exist in either the source or the destination
+      bucket
+    - Source and destination bucket must exist
+    - User must be authenticated with appropriate permissions """
+
+import errno
+import os
+import subprocess
+from typing import List
+import hailtop.batch as hb
+
+
+def batch_move_files(
+    batch: hb.batch,
+    files: List[str],
+    source_prefix: str,
+    destination_prefix: str,
+    docker_image: str = None,
+    key: str = None,
+) -> List:
+    """This script takes a Hail.Batch workflow and a list of files to
```

Comment:
Just a side note: for the main description of a function, I'd recommend keeping it short and simple, without describing individual inputs and outputs or how the function is implemented. The input and outputs are covered below anyway.

Reply:
Have updated :D